raven/src/main/java/org/opencadc/raven/RavenInitAction.java

```diff
@@ -235,4 +282,91 @@ static MultiValuedProperties getConfig() {
             throw new IllegalStateException("invalid config: failed to load SQLGenerator: " + cname);
         }
     }
+
+    private class AvailabilityCheck implements Runnable {
+
+        private Set<StorageSite> sites;
+        private Map<URI, SiteState> siteStaties;
+        private Map<URI, Availability> siteAvailabilities;
+        
+        public AvailabilityCheck(Set<StorageSite> sites) {
+            this.sites = sites;
+            this.siteStaties = new HashMap<>(sites.size());
+            this.siteAvailabilities = new HashMap<>(sites.size());
+            try {
+                Context initialContext = new InitialContext();
+                initialContext.bind(JNDI_AVAILABILITY_KEY, this.siteAvailabilities);
+            } catch (NamingException e) {
+                throw new IllegalStateException(String.format("unable to bind {} to initial context: %s", 
+                                                JNDI_AVAILABILITY_KEY, e.getMessage()), e);
+            }
+        }
+
+        @Override
+        public void run() { 
+            while (true) {
+                for (StorageSite site: this.sites) {
+                    try {
+                        URI resourceID = site.getResourceID();
+                        SiteState siteState = this.siteStaties.get(resourceID);
+                        if (siteState == null) {
+                            siteState = new SiteState(true, 0);
+                        }
+                        boolean minDetail = siteState.isMinDetail();
+                        Availability availability = getAvailability(resourceID, minDetail);
+                        siteState.available = availability.isAvailable();
+                        this.siteStaties.put(resourceID, siteState);
+                        this.siteAvailabilities.put(resourceID, availability);
+                        Context initContext = new InitialContext();
```

Comment:
I don't think you need to rebind the Map<URI,Availability> every time. It is a normal java object and you should be able to modify it without further interaction with JNDI (since you maintain a local reference).

Reply:
Removed rebinding the availability Map every time it's updated.